sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -86,8 +77,27 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
   private var eglPrepared = false
   private var renderNotSupported = false
 
-  internal var fpsChangedListener: OnFpsChangedListener? = null
-  private var timeElapsed = 0L
+  // could not be volatile as setter method is synchronized
+  internal var fpsChangedListener by Delegates.observable<OnFpsChangedListener?>(null) { _, old, new ->
+    new?.let {
+      if (old != it) {
+        /**
+         * Consider setting [OnFpsChangedListener] as non-render event to make sure that
+         * it does not get dropped if user has set it right after MapView creation before render thread
+         * is actually fully prepared for rendering.
+         */
+        postNonRenderEvent(
+          RenderEvent(
+            { fpsManager.fpsChangedListener = it },
```

Comment:
Does this mean previous `fpsManager.fpsChangedListener` will remain if null is set?